### ISSUE TITLE
Use of Switch.pm

### ISSUE THREAD

**Issue by apcraig** [E3SM-Project/E3SM#4507]
As a follow up to #4240, I'm wondering if Switch should be removed from the perl scripts and those sections recoded.  This is not a particularly difficult task.  A recent upgrade on a machine I'm supporting E3SM on did not come with Switch (it had been installed earlier, just like cori).  It could be a while for the admins to install Switch and while I could install it myself, that doesn't necessarily help other users on the same machine.  As a work-around, I'm recoding the Switch sections that cause problems for us and working on a branch others can use.

My sense, looking around the web for solutions, is that Switch might even be deprecated in perl now, although that's not clear to me.  In general, I favor reducing system requirements where they can easily be avoided.  It seems like the community would benefit by getting rid of Switch all-together in the perl in E3SM if it just makes porting more difficult.

So far, I have coded around it in components/mpas-seaice/bld/build-namelist and components/mpas-ocean//bld/build-namelist with if's.  I have not attempted a comprehensive search of all places it occurs.  

Just a suggestion.

**Comment by rljacob** [comment]
We ultimately want to rewrite all the perl in python but just removing Switch.pm dependence in existing perl would definitely help.

**Comment by ndkeen** [comment]
Yes both Switch and libxml are common problems on machines (both new ones as well as those recently back from an upgrade) and it would be a great improvement if these 2 dependencies were removed in some way. Although, I'm getting pretty good at finding work-arounds.

```
#! /usr/bin/perl

use strict; 
use warnings;
use Switch;
use XML::LibXML;

1;
```
